tracer/src/Datadog.Trace/PlatformHelpers/HostMetadata.cs

```diff
@@ -0,0 +1,153 @@
+
+        /// <summary>
+        /// Gets the name of the host on which the code is running
+        /// Returns <c>null</c> if the host name can not be determined
+        /// </summary>
+        public string Hostname { get; }
+
+        /// <summary>
+        /// Gets the name of the kernel, e.g. Linux
+        /// Returns <c>null</c> if it can not be determined
+        /// </summary>
+        public string KernelName { get; }
+
+        /// <summary>
+        /// Gets the release name of the kernel, e.g. 3.2.0-4-686-pae
+        /// Returns <c>null</c> if it can not be determined
+        /// </summary>
+        public string KernelRelease { get; }
+
+        /// <summary>
+        /// Gets the version number of the kernel, e.g. #1 SMP Debian 3.2.63-2+deb7u2
+        /// Returns <c>null</c> if it can not be determined
+        /// </summary>
+        public string KernelVersion { get; }
+
+        // internal for testing
+        internal static void ParseKernel(string fullVersion, out string kernel, out string kernelRelease, out string kernelVersion)
+        {
+            kernel = null;
+            kernelRelease = null;
+            kernelVersion = null;
+
+            if (string.IsNullOrEmpty(fullVersion))
+            {
+                return;
+            }
+
+            var firstWord = fullVersion.IndexOf(value: " ", StringComparison.OrdinalIgnoreCase);
+            if (firstWord < 0)
+            {
+                return;
+            }
+
+            kernel = fullVersion.Substring(0, firstWord);
+
+            var releaseStart = fullVersion.IndexOf("version ", StringComparison.OrdinalIgnoreCase);
+            if (releaseStart < 0)
+            {
+                return;
+            }
+
+            releaseStart += 8; // "version ".Length
+
+            var releaseEnd = fullVersion.IndexOf(" ", startIndex: releaseStart, StringComparison.OrdinalIgnoreCase);
+            if (releaseEnd < 0)
+            {
+                return;
+            }
+
+            kernelRelease = fullVersion.Substring(releaseStart, releaseEnd - releaseStart);
+
+            var versionIndex = fullVersion.LastIndexOf("#", StringComparison.OrdinalIgnoreCase);
+            if (versionIndex < releaseEnd)
+            {
+                return;
+            }
+
+            kernelVersion = fullVersion.Substring(versionIndex);
+        }
+
+        private static void TryGetKernelInformation(out string kernel, out string kernelRelease, out string kernelVersion)
+        {
+            try
+            {
+                if (File.Exists(@"/proc/version"))
+                {
+                    // e.g. Linux version 5.10.60.1-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Wed Aug 25 23:20:18 UTC 2021
+                    var fullVersion = File.ReadAllText("/proc/version");
+
+                    ParseKernel(fullVersion, out kernel, out kernelRelease, out kernelVersion);
```

Comment:
Shouldn't this be followed by a return statement? It seems like the results will just be overwritten at the end of the method

Reply:
🤦‍♂️ yes, originally I had this returning a boolean, and I screwed up in the refactor. Thanks!

Reply:
And of course I was wondering why the tests don't fail: because I don't test this method, just the `ParseKernel()` method 😩

Reply:
Fixed in a new pr https://github.com/DataDog/dd-trace-dotnet/pull/2104